crates/sui-network/src/state_sync/metrics.rs

```diff
@@ -1,14 +1,22 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+use mysten_metrics::histogram::Histogram;
 use prometheus::{register_int_gauge_with_registry, IntGauge, Registry};
 use std::sync::Arc;
+use std::time::Duration;
 use sui_types::messages_checkpoint::CheckpointSequenceNumber;
 use tap::Pipe;
 
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub(super) struct Metrics(Option<Arc<Inner>>);
 
+impl std::fmt::Debug for Metrics {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        fmt.debug_struct("Metrics").finish()
+    }
+}
+
```

Comment:
why is this needed?

Reply:
Histogram doesn't have a Debug impl, so I had to remove it from Inner.